src/librustc/ty/mod.rs

```diff
@@ -275,6 +275,22 @@ impl ImplOrTraitItemId {
     }
 }
 
+#[derive(Clone, Debug, PartialEq, Eq, Copy)]
+pub enum Visibility {
+    Public, //< Visible everywhere
+    Restricted(NodeId), //< Visible only in the given module
+    Private, //< Visible nowhere
```

Comment:
Ah, now I see, `Visibility::Private` is for restricted items from other crates so you can use local `NodeId` in `Visibility::Restricted` and also checks items from other crates quickly. Could you add comments on `Private`/`Restricted` describing this scheme?